src/libtiled/varianttomapconverter.cpp

```diff
@@ -184,6 +185,20 @@ Properties VariantToMapConverter::toProperties(const QVariant &propertiesVariant
     return properties;
 }
 
+Components VariantToMapConverter::toComponents(const QVariant &componentsVariant) const
+{
+    Components components;
+
+    for (QVariant component : componentsVariant.toList()) {
```

Comment:
When using a range-based for with a Qt container, it is important to make sure it is `const` otherwise a non-const iterator is used which can (and will in this case) cause a detach (copy of the implicitly shared data). In addition, we should iterate with `const QVariant &` to avoid additional reference counting:
```suggestion
    const auto componentsVariantList = componentsVariant.toList();
    for (const QVariant &component : componentsVariantList) {
```